### DISCUSSION TITLE
CI/CD tool summary - gocd

### DISCUSSION THREAD

**Discussion by czerniga** [gocd/gocd#10660] [General]
Hi!
I am the author of the CI/CD tool summary: [Find your best CI/CD Tool](https://www.czerniga.it/2022/03/27/find-your-best-ci-cd-tool/) I am planning to update this report soon and I wonder if you would like to verify the information contained therein regarding your tools.
 
During the creation of this report, I tried to find all the necessary information. However, I think that you as the author of this tool have the most verified information.

I would appreciate your help in this matter.

**Comment by chadlwilson** [comment]
Hi @czerniga!

My suggestions for GoCD would be:

> **Platform**: `On-premise & cloud`

While you can deploy GoCD on the cloud in whatever provider you like, that's true for all tools marked as "On-Premise", including Concourse, Bamboo etc. Some raw on virtual machines, some on containers, some on Kubernetes...

GoCD doesn't have a Cloud **SaaS** or "managed" offering like some of the others (Travis, GitLab etc). If your intent is to distinguish "run yourself" vs "managed cloud offering" you might want to tweak the language here and avoid referring to "on-premise"? Otherwise GoCD is more like Bamboo in that sense, except with better container/Kubernetes support (IIRC).

> **External DB Needed**: `Recommended: MySQL, H2, PostgreSQL`

Suggest: `Optional, Recommended for larger installs: PostgreSQL, MySQL

- I probably wouldn't mention **H2** here, because H2 is essentially the default Java-based internal bundled in-memory database, with on-disk storage (not _external_). Users don't have to set it up or manage it, just back it up alongside the rest of the config, and GoCD manages zipping a DB export for backups etc.
- PostgreSQL is more widely testing/validated than MySQL, so would probably order it first ;-)

> **Version control integration**: `Git, Mercurial, SVN, TFS, Perforce`

Suggest: `Git, Mercurial, SVN, TFS/Azure DevOps Server, Perforce`

While it's still referred to as TFS internally in GoCD, it's managed under the "Azure DevOps Server" brand on Microsoft side now.

> **Plugin Source**: `GitHub`

Suggest :`Internal Store/GitHub`

Technically all plugins are loaded from disk (`Internal Store`), however the container builds have some magic that will download them for you at container start and put them in the right place. Most plugins are distributed via GitHub, but they don't have to be.

> **GitHub/Azure AD Authentication**: `GitHub/Azure AD`

You can definitely use GitHub-based auth, but I'm not sure specifically about Azure AD, although it definitely supports LDAP via plugin, so possibly that is good enough.

> **Best for**: `For teams that want a free and open source solution with parallel execution`

This is obviously subjective, and I am biased, but I would say "For teams that want a free and open source solution with advanced pipeline dependency management and workflow visualisation". Maybe a bit salesy 😉

**Reply by czerniga** [reply]
Thank you for your quick and comprehensive reply !
